zboss: Update ZBOSS version to stabilized (09_10_2020)

sebastiandraus opened the pull request:
- ZBOSS sources: ada73ece9d90e7cb467f5e47f4be048651f36031

NordicBuilder commented:
**Some checks failed. Please fix and resubmit.**

## Identity/Emails issues

42ca07be05bdf990d57331fba158961b80e67a67: author email (Draus, Sebastian <sebastian.draus@nordicsemi.no>) does not follow the syntax: First Last <email>.



*Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages.*

tomchy commented:
Is it really that old ZBOSS (07/07/2020)?

tomchy commented on zboss/zboss_release_notes/release_notes.rst:
```diff
@@ -0,0 +1,58 @@
+.. _zboss_changelog:
```
What do you think about using `doc` directory instead of `zboss_release_notes`?

tomchy commented on zboss/zboss_release_notes/release_notes.rst:
```diff
@@ -0,0 +1,58 @@
+Changelog
+#########
+
+All notable changes to this project are documented in this file.
```
Missing notes:
 - Removal of the `zb_void_t` type.
 - Fixing the factory new status issue.
 - Callback passed as parameter in APSME API.
 - Is `ZB_JOINING_LIST_SUPPORT` and `CONFIG_ZB_ENHANCED_BEACON_SUPPORT` feature mentioned somewhere or is it a bug in the release script?

tomchy replied:
Just to recap: we agreed to add three release notes. The `ZB_JOINING_LIST_SUPPORT` and `CONFIG_ZB_ENHANCED_BEACON_SUPPORT` are new macros, that allows to enable/disable functionalities, that were already included in previous nrfxlib's ZBOSS.

tomchy reviewed (changes requested):
There are several questionable changes, that has to be fixed or documented. I will try to help with them.

tomchy commented:
@greg-fer Could you look at the release notes file?

tomchy commented on zboss/CHANGELOG.rst:
```diff
@@ -0,0 +1,85 @@
+=======
+
+* First release with updated versioning scheme, that refers to the ZBOSS release naming convention.
+* Replaced zb_void_t with void is stack core.
```
`is`->`in`

tomchy commented on zboss/include/osif/zb_vendor.h:
```diff
@@ -244,6 +279,8 @@
 /* ZBOSS stack options non-configurable via Kconfig. */
 
+/* Remove upper layers when building MAC tests. */
+
```
This looks interesting, but it is correct. The `unifdef` tool removed the `#ifndef ZB_MAC_TESTING_MODE`. I think we may keep it as-is for now :slightly_smiling_face:

tomchy commented:
Please remember to add a PR to sdk-nrf repository, that will be used to verify updated libraries.

greg-fer commented:
I need to discuss this with @b-gent , as he's coordinating most of nrfxlib updates. I have a doubt the changelog should actually be added to ZBOSS stack docs and not documented within NCS, also because of different release approach (this PR is based on ZBOSS versioning, while other changelogs in nrfxlib are based on NCS releases).

b-gent commented:
I'm going to remove myself from the reviewers list, we discussed with @greg-fer on how to proceed.

greg-fer commented on zboss/CHANGELOG.rst:
```diff
@@ -0,0 +1,85 @@
+
+All notable changes to this project are documented in this file.
+
+ZBOSS 09/10/2020 (v3.3.0.3+09_10_2020)
```
I suggest we go for `nRF Connect SDK v1.4.0` for this header, and merge all other changes under this one.
That is: remove all ZBOSS headers, and keep changes under one v1.4.0 header.
If there are any items that are related to each other between different ZBOSS releases, please merge them together into one item.

greg-fer commented on zboss/README.rst:
```diff
@@ -6,3 +6,9 @@ ZBOSS Zigbee stack
 The |NCS|'s Zigbee stack uses ZBOSS – a portable, high-performance Zigbee software protocol stack that allows for interoperability, customizing, testing, and optimizing of your Zigbee solution.
 
 For detailed documentation of the ZBOSS API and instructions on how to use it, see `ZBOSS API documentation`_.
+
```
As per Teams conversation with @wbober , please add a `ZBOSS configuration` section with a mention of the header file:
`For an overview of the default ZBOSS configuration, see :file:`<name of the file>`.
You can also provide information about what this configuration is about and link to configuring Zigbee in NCS...

sebastiandraus commented:
- Removed stack related changes from CHANGELOG
- Mentioned ZBOSS libraries configuration in zboss README
- squashed and rebased on newest master branch

tomchy commented on zboss/README.rst:
```diff
@@ -6,3 +6,21 @@ ZBOSS Zigbee stack
+
+The ZBOSS stack is distributed as a set of precompiled linkable libraries.
+There are two variants of the libraries that support different Zigbee device roles, each variant having its own configuration.
+For an overview of the ZBOSS library configuration, see the following files:
```
This is not just an overview, but a complete list of configs.

wbober replied:
@tomchy @greg-fer If changing this triggers CI run I'd prefer to keep it as is.

rlubos reviewed (approved):
LGTM
